controllers/suite_test.go

```diff
@@ -177,6 +177,22 @@ var _ = BeforeSuite(func() {
 		}
 		Expect(k8sClient.Create(ctx, rookConfigMap)).ShouldNot(HaveOccurred())
 
+		var grafanaDatasources struct {
+			Datasources [1]struct {
+				BasicAuthPassword string `json:"basicAuthPassword"`
+				BasicAuthUser     string `json:"basicAuthUser"`
+			} `json:"datasources"`
+		}
+		grafanaDatasources.Datasources[0].BasicAuthPassword = "password"
+		grafanaDatasources.Datasources[0].BasicAuthUser = "internal"
+		grafanaDatasourceSecret := &corev1.Secret{}
+		grafanaDatasourceSecret.Name = testGrafanaFederateSecretName
+		grafanaDatasourceSecret.Namespace = testOpenshiftMonitoringNamespace
+		grafanaDatasourceSecret.Data = map[string][]byte{
+			"prometheus.yaml": utils.ToJsonOrDie(grafanaDatasources),
+		}
+		Expect(k8sClient.Create(ctx, grafanaDatasourceSecret)).ShouldNot(HaveOccurred())
```

Comment:
Why was this moved here?

Reply:
This resource should by default exist in the monitoring namespace and it shouldn't be created while the test is running. So added it to BeforeSuite which ensures that the resource exists in the monitoring namespace.

Reply:
How did it work before?

Reply:
Before we were creating the secret in monitoring namespace in managedocs_controller_test.go file just before we check for k8sMetricsServiceMonitorAuthSecret
https://github.com/red-hat-storage/ocs-osd-deployer/blob/main/controllers/managedocs_controller_test.go#L1142-L1143

Reply:
The old way makes more sense as it is the trigger for the test. 
Right now you do not actually have a test as you do not have a trigger for it. 

Let's revert.